commodore/dependency_templater.py

```diff
@@ -85,7 +85,26 @@ def _base_from_existing(cls, config: Config, path: Path, deptype: str):
         t._target_dir = path
         t.output_dir = path.absolute().parent
 
-        t._initialize_from_cookiecutter_args(cookiecutter_args)
+        # We pass the cookiecutter args dict to `_initialize_from_cookiecutter_args()`.
+        # Because Python dicts are passed by reference, the function can simply add
+        # missing args into the dict and return `True` to cause us to write back and
+        # commit the updated `.cruft.json` data.
+        update_cruft_json = t._initialize_from_cookiecutter_args(cookiecutter_args)
```

Comment:
This is a bit too much magic for my taste, but I don't have a better solution. But I'd propose that at least add a doc string to `_initialize_from_cookiecutter_args` so that when someone looks at the method (or tried to extend the templater) they know why it returns a boolean and what it means without having to read this method.

Reply:
I'm not particularly happy with the amount of magic here either, but less magic would mean that we need to have a bunch of additional repetitive code. I'll add a docstring to explain the magic which happens.